### PR TITLE
ekf2: fill pose and vel covariance in odometry

### PR DESCRIPTION
**Pull request by dagar**
Resurrecting https://github.com/PX4/PX4-Autopilot/pull/11092.

### CODE REVIEW

**Comment by LorenzMeier** [comment]
@bresch Could you review?

File: src/modules/ekf2/EKF2.cpp [hunk 1]
```diff
@@ -843,6 +843,56 @@ void EKF2::PublishLocalPosition(const hrt_abstime &timestamp)
...
+	matrix::Matrix<float, 3, 4> G;
+
+	// quaternion components
+	float q1 = q(0);
+	float q2 = q(1);
+	float q3 = q(2);
+	float q4 = q(3);
+
+	// numerator components
+	float n1 =  2 * q1 * q2 + 2 * q2 * q4;
```
Comment (priseborough) [inline src/modules/ekf2/EKF2.cpp:860]: Where did these come from?

Comment (priseborough) [reply src/modules/ekf2/EKF2.cpp:860]: I will see if this derivation can be added to our existing sympy script.

File: src/modules/ekf2/EKF2.cpp [hunk 2]
```diff
@@ -843,6 +843,56 @@ void EKF2::PublishLocalPosition(const hrt_abstime &timestamp)
...
+	float n3 =  2 * q1 * q4 + 2 * q2 * q3;
+	float n4 = -2 * q3 * q3 - 2 * q4 * q4 + 1;
+	float n5 =  2 * q1 * q3 + 2 * q2 * q4;
+	float n6 = -2 * q1 * q2 - 2 * q2 * q4;
+	float n7 = -2 * q1 * q4 - 2 * q2 * q3;
+
+	// Protect against division by 0
+	float d1 = n1 * n1 + n2 * n2;
+	float d2 = n3 * n3 + n4 * n4;
+	d1 = math::max(0.0f, d1);
```
Comment (priseborough) [inline src/modules/ekf2/EKF2.cpp:871]: This doesn't protect against /0 where we divide by d1. Likewise for d2.

File: src/modules/ekf2/EKF2.cpp (hunk 1 above)
Comment (priseborough) [reply src/modules/ekf2/EKF2.cpp:860]: See https://github.com/priseborough/PX4-Autopilot/tree/pr-ekf2_derivation_update for a symbolic derivation

Comment (dagar) [reply src/modules/ekf2/EKF2.cpp:860]: Nuno's original PR for context. https://github.com/PX4/PX4-Autopilot/pull/11092

**Comment by dagar** [comment]
Rolled into https://github.com/PX4/PX4-Autopilot/pull/19966.

**Pull request closed without merging**